src/shared/lib/OncoKbUtils.ts

```diff
@@ -0,0 +1,146 @@
+        highestLevelOnly: true,
+        levels: ['LEVEL_1', 'LEVEL_2A', 'LEVEL_2B', 'LEVEL_3A', 'LEVEL_3B', 'LEVEL_4', 'LEVEL_R1'],
+        queries: queryVariants,
+        source: "cbioportal"
+    }
+}
+
+export function generateQueryVariant(hugoSymbol:string,
+                                     mutationType:string,
+                                     proteinChange:string,
+                                     proteinPosStart:number,
+                                     proteinPosEnd:number,
+                                     tumorType:string): Query
+{
+    return {
+        id: generateQueryVariantId(hugoSymbol, mutationType, proteinChange, tumorType),
+        hugoSymbol,
+        tumorType,
+        alterationType: "MUTATION",
+        entrezGeneId: 0,
+        alteration: proteinChange,
+        consequence: consequenceConverter(mutationType),
+        proteinStart: proteinPosStart,
+        proteinEnd: proteinPosEnd,
+    };
+}
+
+export function generateQueryVariantId(hugoSymbol:string,
+                                       mutationType:string,
+                                       proteinChange:string,
+                                       tumorType:string): string
+{
+    return `${hugoSymbol}_${proteinChange}_${tumorType}_${mutationType}`;
+}
+
+function getLevel(level:string):string|null
+{
+    if (level)
+    {
+        const matchArray = level.match(/LEVEL_(R?\d[AB]?)/);
+
+        if (matchArray && matchArray.length >= 2) {
+            return matchArray[1];
+        }
+        else {
+            return level;
+        }
+    }
+    else {
+        return null;
+    }
+}
+
+export function oncogenicImageClassNames(oncogenic:string,
+                                         isVUS:boolean,
+                                         highestSensitiveLevel:string,
+                                         highestResistanceLevel:string):string[]
+{
+    let classNames = ["", ""];
+
+    const oncogenicToIcon:{[oncogenic:string] : string} = {
+        'Likely Neutral': 'likely-neutral',
+        'Unknown': 'unknown-oncogenic',
+        'Inconclusive': 'unknown-oncogenic',
+        'Likely Oncogenic': 'oncogenic',
+        'Oncogenic': 'oncogenic',
+    };
+
+    const sl = getLevel(highestSensitiveLevel);
+    const rl = getLevel(highestResistanceLevel);
+
+    if (!rl && sl)
+    {
+        classNames[0] = 'level' + sl;
+    }
+    else if (rl && !sl)
+    {
+        classNames[0] = 'level' + rl;
+    }
+    else if (rl && sl)
+    {
+        classNames[0] = 'level' + sl + 'R';
+    }
+
+    classNames[1] = oncogenicToIcon[oncogenic] || "no-info-oncogenic";
+
+    if(classNames[1] === 'no-info-oncogenic' && isVUS)
+    {
+        classNames[1] = 'vus';
+    }
+
+    return classNames;
+}
+
+/**
+ * Convert cBioPortal consequence to OncoKB consequence
+ *
+ * @param consequence cBioPortal consequence
+ * @returns
+ */
+export function consequenceConverter(consequence:string) {
```

Comment:
sounds better. I just copied this from the legacy code base without paying much attention to its name :)